File: adapters/dart/lib/src/precompiled/target.dart
import 'dart:io';

String detectHostTargetTriple() {
  final arch = _detectArch();
  switch (Platform.operatingSystem) {
    case 'macos':
      return arch == 'aarch64'
          ? 'aarch64-apple-darwin'
          : 'x86_64-apple-darwin';
    case 'linux':
      return arch == 'aarch64'
          ? 'aarch64-unknown-linux-gnu'
          : 'x86_64-unknown-linux-gnu';
    case 'windows':
      return arch == 'aarch64'
          ? 'aarch64-pc-windows-msvc'
          : 'x86_64-pc-windows-msvc';
    case 'android':
      return switch (arch) {
        'aarch64' => 'aarch64-linux-android',
        'arm' => 'armv7-linux-androideabi',
        'x86' => 'i686-linux-android',
        _ => 'x86_64-linux-android',
      };
    case 'ios':
      return arch == 'aarch64'
          ? 'aarch64-apple-ios'
          : 'x86_64-apple-ios';
    default:
      return 'x86_64-unknown-linux-gnu';
  }
}

String _detectArch() {
  final env = Platform.environment;
  final raw = [
    env['PROCESSOR_ARCHITECTURE'],
    env['PROCESSOR_IDENTIFIER'],
    env['HOSTTYPE'],
    env['MACHTYPE'],
    Platform.version,
    Platform.operatingSystemVersion,
  ].whereType<String>().join(' ').toLowerCase();

  if (raw.contains('aarch64') || raw.contains('arm64')) {
    return 'aarch64';
  }
  if (raw.contains('x86_64') || raw.contains('amd64') || raw.contains('x64')) {
    return 'x86_64';
  }
  if (raw.contains('i686') || raw.contains('i386') || raw.contains('x86')) {
    return 'x86';
  }
  if (raw.contains('arm')) {
    return 'arm';
  }
  return 'x86_64';
}

File: adapters/python/src/xforge/target.py
"""
Detect host Rust target triple (e.g. aarch64-apple-darwin, x86_64-unknown-linux-gnu).
"""
import platform
import sys


def detect_host_target_triple() -> str:
    machine = platform.machine().lower()
    system = sys.platform

    arch = "x86_64"
    if "aarch64" in machine or "arm64" in machine:
        arch = "aarch64"
    elif "x86_64" in machine or "amd64" in machine or "x64" in machine:
        arch = "x86_64"
    elif "i686" in machine or "i386" in machine or "x86" in machine:
        arch = "x86"
    elif "arm" in machine:
        arch = "arm"

    if system == "darwin":
        return f"{arch}-apple-darwin"
    if system == "linux":
        return f"{arch}-unknown-linux-gnu"
    if system == "win32":
        return f"{arch}-pc-windows-msvc"
    # Fallback
    return "x86_64-unknown-linux-gnu"
